binding-library/java/src/main/java/com/microsoft/azure/functions/signalr/annotation/SignalRConnectionInfoInput.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.functions.signalr.annotation;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.ElementType;
+
+import com.microsoft.azure.functions.annotation.CustomBinding;
+
+/**
+ * <p>Place this on a parameter to obtain a SignalRConnectionInfo object.
+ * The parameter type can be one of the following:</p>
+ *
+ * <ul>
+ *     <li>SignalRConnectionInfo type</li>
+ * </ul>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@CustomBinding(direction = "in", name = "connectionInfo", type = "SignalRConnectionInfo")
```

Comment:
Should `name` have a value? In @pragnagopa's sample it is an empty string: https://github.com/Azure/azure-functions-java-worker/pull/271#issue-252885238

Reply:
It could. In maven function build logic, memeber name() is first priority, then name in customBinding. Just want to convenient user of assign name. 
If we remove the default "", it means user must assign the name and name in CustomBinding is actually useless. 

---
In reply to: [259248519](https://github.com/Azure/azure-functions-signalrservice-extension/pull/48#discussion_r259248519) [](ancestors = 259248519)

Reply:
Do you suggest to make it required and remove default ""?

---
In reply to: [259248519](https://github.com/Azure/azure-functions-signalrservice-extension/pull/48#discussion_r259248519) [](ancestors = 259248519)

Reply:
Looking at a few bindings that are in the library, they all have `name` as required (even for output bindings). I think we should stay consistent and remove default from this one.

Reply:
Sure. Will update, for both Input/Output.

---
In reply to: [259255874](https://github.com/Azure/azure-functions-signalrservice-extension/pull/48#discussion_r259255874) [](ancestors = 259255874)